### ISSUE TITLE
Clang and wasm

### ISSUE THREAD

**Issue by ifuchs** [holzschu/a-shell#38]

**Comment by ifuchs** [comment]
I am playing around with clang and wasm, using some c tutorial files and I cannot get scans to work. Is this a limitation of the implementation? What am I missing? Here is the test file:

`include <stdio.h>

int main () {
   char str1[20], str2[30];
   printf("Enter name: ");
   scanf("%s", str1);
   printf("Enter your website name: ");
   scanf("%s", str2);
   printf("Entered Name: %s\n", str1);
   printf("Entered Website:%s", str2);
   return(0);
}
~                                                                                                                                 
~                                                                                                                                 
"scantest.c" [New] 17L, 275C written                                                                  
$ clang scantest.c
$ wasm a.out
Enter name: Enter your website name: Entered Name: 
Entered Website:$  `

**Comment by holzschu** [comment]
Yes, this is known limitation. For the time being, wasm listens to stdin if it is redirected, but not if it comes from the keyboard. So `cat name website_name > wasm a.out` will work, but not what you tried. I'll keep that in mind as a possible use.

**Comment by ifuchs** [comment]
Having a way to get input for C and C++ programs would  certainly make it
more useful.

**Comment by holzschu** [comment]
That will probably come with the ability to open files (also something really useful).

**Comment by ifuchs** [comment]
Now that you’ve added the file capability, perhaps you can also add input from stdin?

**Comment by holzschu** [comment]
I'm trying...
